homekit/StateManager.js

```diff
@@ -1,623 +1,1039 @@
+					easyDebugInfo(device.name, '(GET) - Current Temperature:', currentTemp + 'ºC')
 				}
 
 				callback(null, currentTemp)
 			},
 
+			/** @param {homebridge.CharacteristicGetCallback} callback */
 			CoolingThresholdTemperature: (callback) => {
-				const targetTemp = sanitize(device.HeaterCoolerService, 'CoolingThresholdTemperature', device.state.targetTemperature)
+				if (!(device.state instanceof Classes.InternalAcState)) {
+					easyDebugInfo(device.name, `(GET) - Target Cooling Temperature: ${device.state} is not an instance of InternalAcState!`)
+
+					return
+				}
+
+				if (!(device instanceof AirConditioner)) {
+					easyDebugInfo(device.name, `(GET) - Target Cooling Temperature: ${device.name} is not an instance of AirConditioner!`)
+
+					return
+				}
+
+				const targetTemp = this.sanitize(device.HeaterCoolerService, 'CoolingThresholdTemperature', device.state.targetTemperature)
 
 				if (device.usesFahrenheit) {
-					log.easyDebug(device.name, '(GET) - Target Cooling Temperature:', toFahrenheit(targetTemp) + 'ºF')
+					easyDebugInfo(device.name, '(GET) - Target Cooling Temperature:', this.toFahrenheit(targetTemp) + 'ºF')
 				} else {
-					log.easyDebug(device.name, '(GET) - Target Cooling Temperature:', targetTemp + 'ºC')
+					easyDebugInfo(device.name, '(GET) - Target Cooling Temperature:', targetTemp + 'ºC')
 				}
 
 				callback(null, targetTemp)
 			},
 
+			/** @param {homebridge.CharacteristicGetCallback} callback */
 			HeatingThresholdTemperature: (callback) => {
-				const targetTemp = sanitize(device.HeaterCoolerService, 'HeatingThresholdTemperature', device.state.targetTemperature)
+				if (!(device.state instanceof Classes.InternalAcState)) {
+					easyDebugInfo(device.name, `(GET) - Target Heating Temperature: ${device.state} is not an instance of InternalAcState!`)
+
+					return
+				}
+
+				if (!(device instanceof AirConditioner)) {
+					easyDebugInfo(device.name, `(GET) - Target Heating Temperature: ${device.name} is not an instance of AirConditioner!`)
+
+					return
+				}
+
+				const targetTemp = this.sanitize(device.HeaterCoolerService, 'HeatingThresholdTemperature', device.state.targetTemperature)
 
 				if (device.usesFahrenheit) {
-					log.easyDebug(device.name, '(GET) - Target Heating Temperature:', toFahrenheit(targetTemp) + 'ºF')
+					easyDebugInfo(device.name, '(GET) - Target Heating Temperature:', this.toFahrenheit(targetTemp) + 'ºF')
 				} else {
-					log.easyDebug(device.name, '(GET) - Target Heating Temperature:', targetTemp + 'ºC')
+					easyDebugInfo(device.name, '(GET) - Target Heating Temperature:', targetTemp + 'ºC')
 				}
 
 				callback(null, targetTemp)
 			},
 
+			/** @param {homebridge.CharacteristicGetCallback} callback */
 			TemperatureDisplayUnits: (callback) => {
-				log.easyDebug(device.name, '(GET) - Temperature Display Units:', device.temperatureUnit)
+				if (!(device instanceof AirConditioner || device instanceof RoomSensor)) {
```

Comment:
[Tempature Sensor](https://developers.homebridge.io/#/service/TemperatureSensor) doesn't support display unit characteristic

```suggestion
				if (!(device instanceof AirConditioner)) {
```